### PR TITLE
fix(event_sources): Update CodePipeline event source to include optional encryption_key field and make user_parameters field optional

### LINKED ISSUE #2117: Bug: CodePipeline Event Data Class incorrectly handles optional User Configuration
### Expected Behaviour

On a [CodePipeline Job Data Class](https://awslabs.github.io/aws-lambda-powertools-python/2.11.0/utilities/data_classes/#codepipeline-job), accessing either `decoded_user_parameters` or `user_parameters` should return `None` if no User Parameters were supplied.

### Current Behaviour

Attempting to call `decoded_user_parameters` or `user_parameters`, when none were supplied, causes a KeyError to occur.

The [AWS CodePipeline Lambda Action](https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-Lambda.html) does not require UserParameters, however the CodePipeline Event Data Class expects it to be set, and raises a KeyError.

### Code snippet

```python
from aws_lambda_powertools.utilities.data_classes import (
    event_source,
    CodePipelineJobEvent,
)
from aws_lambda_powertools.utilities.typing import LambdaContext
from aws_lambda_powertools.logging import Logger
import boto3

CODEPIPELINE = boto3.client("codepipeline")
logger = Logger(service="bug-report", level="DEBUG")

@event_source(data_class=CodePipelineJobEvent)
def handler(event: CodePipelineJobEvent, context: LambdaContext):
    try:
        logger.debug(str(event.raw_event))
        event.decoded_user_parameters
    except Exception as e:
        logger.exception(e, stack_info=True)
        CODEPIPELINE.put_job_failure_result(
            jobId=event.get_id, failureDetails={"type": "JobFailed", "message": str(e)}
        )
        raise e

    CODEPIPELINE.put_job_success_result(jobId=event.get_id)
```

### Possible Solution

See #2113

### Steps to Reproduce

1. Create a CodePipeline with a AWS Lambda Build Step, without User Parameters. image

2. Write Lambda to access `event.decoded_user_parameters`

3. Note incoming JSON event image

```
Traceback (most recent call last):
  File \"/var/task/pipeline_action.py\", line 22, in handler
    check_inputs(event)
  File \"/var/task/pipeline_action.py\", line 119, in check_inputs
    elif \"TemplateFile\" not in event.decoded_user_parameters:
  File \"/var/task/aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py\", line 165, in decoded_user_parameters
    return self.data.action_configuration.configuration.decoded_user_parameters
  File \"/var/task/aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py\", line 27, in decoded_user_parameters
    self._json_data = json.loads(self.user_parameters)
  File \"/var/task/aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py\", line 21, in user_parameters
   return self[\"UserParameters\"]
  File \"/var/task/aws_lambda_powertools/utilities/data_classes/common.py\", line 17, in __getitem__
    return self._data[key]
KeyError: 'UserParameters'
```

### AWS Lambda Powertools for Python version

latest

### AWS Lambda function runtime

3.8

### Packaging format used

PyPi

### Debugging logs

_No response_

### PR DESCRIPTION
**Pull request by neilramsay**
**Issue number:** #2117 

## Summary
The AWS CodePipeline Lambda Action does not require UserParametersto be set, and therefore will be missing from the `action_configuration`.
https://docs.aws.amazon.com/codepipeline/latest/userguide/action-reference-Lambda.html

### Changes

> Make CodePipelineJobEvent user_parameters Optional (as in Python typing optional)

### User experience

> Prior to change, calling event.user_parameters would result in an exception due to
> the "UserParameters" key not being in the Action Configuration data class.

> Post change, this will return `None`. When using Python type checking, this will inform the developer they need to handle the case where the User Parameters are not present.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [ ] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

Not explicitly; runtime wise, this will not break any existing code that works.
It will create additional errors for those performing type checking, and haven't handled this case.

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by rubenfonseca** [comment]
Thank you so much for your contribution @neilramsay! As stated in our [CONTRIBUTING guide](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/CONTRIBUTING.md) all pull requests require an Issue first.

We'll try to create one for you as soon as possible.

**Comment by neilramsay** [comment]
@rubenfonseca I'll create issue.

**Comment by leandrodamascena** [comment]
> @rubenfonseca I'll create issue.

Thank you for creating the issue @neilramsay! I'll review this PR till tomorrow.

File: aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py [hunk 1]
```diff
@@ -97,6 +97,16 @@ def expiration_time(self) -> Optional[int]:
...
+class CodePipelineEncryptionKey(DictWrapper):
+    @property
+    def get_id(self) -> str:
+        return self["id"]
+
+    @property
+    def get_type(self) -> str:
+        return self["type"]
+
+
```
Comment (leandrodamascena) [inline aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py:109]: Hi @neilramsay! Looking at the documentation, I see that the `encryptionKey` field was missing. When you configure a codepipeline with a customer-managed KMS key, the codepipeline includes this field in the Lambda Event.

**Review by leandrodamascena — APPROVED** [review]
This PR looks good to me. Do you have any considerations before we merge it @neilramsay?

**Comment by neilramsay** [comment]
@leandrodamascena nothing extra to add to this PR prior to merge.
If I find anything else, I'll raise a new issue+PR.

**Pull request merged**